docarray/array/array/pushpull/file.py

```diff
@@ -0,0 +1,182 @@
+
+            for da_file in da_files:
+                table.add_row(
+                    da_file.stem,
+                    str(datetime.fromtimestamp(int(da_file.stat().st_ctime))),
+                    str(filesize.decimal(da_file.stat().st_size)),
+                )
+
+            Console().print(table)
+
+        return [dafile.stem for dafile in da_files]
+
+    @staticmethod
+    def delete(name: str, missing_ok: bool = False) -> bool:
+        """Delete a DocumentArray from the local filesystem.
+
+        :param name: The name of the DocumentArray to delete.
+        :param missing_ok: If True, do not raise an exception if the file does not exist. Defaults to False.
+        :return: True if the file was deleted, False if it did not exist.
+        """
+        path = PushPullFile._abs_filepath(name)
+        try:
+            path.with_suffix('.da').unlink()
+            return True
+        except FileNotFoundError:
+            if not missing_ok:
+                raise
+        return False
+
+    @staticmethod
+    def push(
+        da: 'DocumentArray',
+        name: str,
+        public: bool,
+        show_progress: bool,
+        branding: Optional[Dict],
+    ) -> Dict:
+        """Push this DocumentArray object to the specified file path.
+
+        :param name: The file path to push to.
+        :param public: Not used by the ``file`` protocol.
+        :param show_progress: If true, a progress bar will be displayed.
+        :param branding: Not used by the ``file`` protocol.
+        """
+        return PushPullFile.push_stream(iter(da), name, public, show_progress, branding)
+
+    @staticmethod
+    def push_stream(
+        docs: Iterator['BaseDocument'],
+        name: str,
+        public: bool = True,
+        show_progress: bool = False,
+        branding: Optional[Dict] = None,
+    ) -> Dict:
+        """Push a stream of documents to the specified file path.
+
+        :param docs: a stream of documents
+        :param name: The file path to push to.
+        :param public: Not used by the ``file`` protocol.
+        :param show_progress: If true, a progress bar will be displayed.
+        :param branding: Not used by the ``file`` protocol.
+        """
+        if branding is not None:
+            logging.warning('branding is not supported for "file" protocol')
+
+        source = _to_binary_stream(
+            docs, protocol='protobuf', compress='gzip', show_progress=show_progress
+        )
+        path = PushPullFile._abs_filepath(name).with_suffix('.da.tmp')
+        if path.exists():
+            raise ConcurrentPushException(f'File {path} already exists.')
+        with open(path, 'wb') as f:
+            while True:
+                try:
+                    f.write(next(source))
+                except StopIteration:
+                    break
+        path.rename(path.with_suffix(''))
+        return {}
+
+    @staticmethod
+    def pull(
+        cls: Type['DocumentArray'],
```

Comment:
it is missleading to call this one `cls` I thought it was a classmethod. Maybe `da_cls` ?